### PR TITLE
RF-52 Make secrets service fail, if it couldn't find some secret

### PR DESCRIPTION
**Pull request by stepanGerasimovich**

### CODE REVIEW

**Review by ilya-tishonok — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
 * MongoUrl (string)
 * MongoReplicasetName (string)
-* MongoServerKeepAlive (0 or 1)
-* MongoUseNewUrlParser (0 or 1)
-* MongoConnectTimeout (milliseconds)
-* MongoSocketTimeout (milliseconds)
+
+Optional secrets (.env variables) are:
+* ProjectName (string, defaults to 'Sample App')
+* Environment (string, defaults to 'production')
```
Comment (vladbronishevsky) [inline README.md:11]: 'development' is safer?

File: lib/defaults.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  ProjectName: 'Sample App',
+  Environment: 'production',
```
Comment (vladbronishevsky) [inline lib/defaults.js:3]: 'development'

File: lib/index.js [hunk 3]
```diff
@@ -2,34 +2,31 @@
...
-};
-
 class AppConfigLoader {
   constructor() {
     try {
       this.getSecretFromService = secretsService.init();
-    } catch(e) {
-      console.log(e);
+    } catch(error) {
+      console.log(error);
```
Comment (vladbronishevsky) [inline lib/index.js:10]: let's avoid 'console.log'
we have a logger or an equivalent

File: lib/index.js [hunk 4]
```diff
@@ -2,34 +2,31 @@
...
 
   async getSecret(key) {
     try {
-      return this.getSecretFromService(key) || DEFAULT[key];
-    } catch(e) {
-      console.log(e);
-      return Promise.reject(e);
+      return this.getSecretFromService(key);
+    } catch(error) {
+      console.log(error);
```
Comment (vladbronishevsky) [inline lib/index.js:20]: similar as above

File: lib/index.js [hunk 5]
```diff
@@ -2,34 +2,31 @@
...
     }
   }
 
   async getSecret(key) {
     try {
-      return this.getSecretFromService(key) || DEFAULT[key];
-    } catch(e) {
-      console.log(e);
-      return Promise.reject(e);
+      return this.getSecretFromService(key);
```
Comment (vladbronishevsky) [inline lib/index.js:18]: move this logic to the service

File: lib/secretsService.js [hunk 6]
```diff
@@ -10,7 +11,25 @@ function init() {
     return initAzureVault();
   }
 
-  return secretName => process.env[secretName];
+  return initProccessEnvSecrets();
+}
+
+function initProccessEnvSecrets() {
```
Comment (vladbronishevsky) [inline lib/secretsService.js:17]: Please, restructure to the class with 2 methods:
- getSecret
- getOptionalSecret

**Review by vladbronishevsky — APPROVED** [review]

**Pull request merged**